internal/client/src/client/client.go

```diff
@@ -377,6 +380,41 @@ func GetWalletAddress() ([]byte, error) {
 	return block.HashSHA256(pubKeyEncoded), nil
 }
 
+func RequestWalletInfo(producerAddr string) (accounts.AccountInfo, error) {
+	var accInfo accounts.AccountInfo
+	walletAddress, err := GetWalletAddress()
+	if err != nil {
+		return accInfo, errors.New("failed to get wallet address: " + err.Error())
+	}
+	var requestInfoMessage []byte
+	requestInfoMessage = append(requestInfoMessage, SecretBytes...)
+	requestInfoMessage = append(requestInfoMessage, 2)
+	requestInfoMessage = append(requestInfoMessage, walletAddress...)
+	conn, err := net.Dial("tcp", producerAddr)
+	if _, err := conn.Write(requestInfoMessage); err != nil {
+		return accInfo, errors.New("failed to send message to producer: " + err.Error())
+	}
+	// Should receive Thank you first
+	buf := make([]byte, 1024)
+	if _, err := conn.Read(buf); err != nil {
+		return accInfo, errors.New("failed to get thank you message: " + err.Error())
+	}
+	// Should receive message next
+	buf = make([]byte, 1024)
+	nRead, err := conn.Read(buf)
+	if err != nil {
+		return accInfo, errors.New("failed to get response message: " + err.Error())
+	}
+	if buf[8] == 1 {
```

Comment:
This is the case that means "producer did not find the given wallet address in the table"